api/client-server/directory.yaml

```
@@ -159,3 +168,70 @@ paths:
             "$ref": "definitions/errors/error.yaml"
       tags:
         - Room directory
+  "/rooms/{roomId}/aliases":
+    get:
+      summary: Get a list of local aliases on a given room.
+      description: |-
+        Get a list of aliases maintained by the local server for the
+        given room.
+
+        This endpoint can be called by users who are in the room (external
+        users receive an ``M_FORBIDDEN`` error response). If the room's
+        ``m.room.history_visibility`` maps to ``world_readable``, any
+        user can call this endpoint.
+
+        Servers may choose to implement additional access control checks here,
+        such as allowing server administrators to view aliases regardless of
+        membership.
+
+        .. Note::
+           Clients are not recommended to display this list of aliases prominently
```

```suggestion
           Clients are recommended not to display this list of aliases prominently
```

The previous wording felt like we weren't actually dissuading clients from doing this.